### PR TITLE
Configure compression and chunk size

### PR DESCRIPTION
**Pull request by baldersheim**
@toregge PR
another part of #14181
This should be wiring in of new configuration and updated interfaces. Should have no semantic changes.

### CODE REVIEW

File: searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp [hunk 1]
```diff
@@ -582,12 +602,12 @@ DomainPart::write(FastOS_FileInterface &file, const Packet::Entry &entry)
     int32_t crc(0);
     uint32_t len(entry.serializedSize() + sizeof(crc));
     nbostream os;
-    os << static_cast<uint8_t>(_defaultCrc);
+    os << static_cast<uint8_t>(Crc::xxh64);
```
Comment (toregge) [inline searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp:605]: Get crc method from _encoding ?

File: searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp [hunk 2]
```diff
@@ -582,12 +602,12 @@ DomainPart::write(FastOS_FileInterface &file, const Packet::Entry &entry)
...
     os << len;
     size_t start(os.size());
     entry.serialize(os);
     size_t end(os.size());
-    crc = calcCrc(_defaultCrc, os.data() + start, end - start);
+    crc = calcCrc(Crc::xxh64, os.data() + start, end - start);
```
Comment (toregge) [inline searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp:610]: Get crc method from _encoding ?

File: searchlib/src/vespa/searchlib/transactionlog/translogserverapp.cpp [hunk 3]
```diff
@@ -24,28 +25,72 @@ TransLogServerApp::TransLogServerApp(const config::ConfigUri & tlsConfigUri,
...
+Encoding::Crc
 getCrc(searchlib::TranslogserverConfig::Crcmethod crcType)
 {
     switch (crcType) {
         case searchlib::TranslogserverConfig::Crcmethod::ccitt_crc32:
-            return DomainPart::ccitt_crc32;
+            return Encoding::Crc::ccitt_crc32;
         case searchlib::TranslogserverConfig::Crcmethod::xxh64:
-            return DomainPart::xxh64;
+            return Encoding::Crc ::xxh64;
```
Comment (toregge) [inline searchlib/src/vespa/searchlib/transactionlog/translogserverapp.cpp:35]: Remove unneeded space ?

Comment (baldersheim) [reply searchlib/src/vespa/searchlib/transactionlog/translogserverapp.cpp]: Fixed

File: searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp (hunk 2 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp]: Fixed

File: searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp]: Fixed

**Pull request merged**